deployment/ccip/changeset/test_helpers.go

```diff
@@ -262,37 +259,105 @@ func NewMemoryEnvironmentWithJobsAndContracts(t *testing.T, lggr logger.Logger,
 	require.NoError(t, err)
-	require.NoError(t, e.Env.ExistingAddresses.Merge(out.AddressBook))
+
 	state, err := LoadOnchainState(e.Env)
 	require.NoError(t, err)
-
-	newAddresses = deployment.NewMemoryAddressBook()
 	tokenConfig := NewTestTokenConfig(state.Chains[e.FeedChainSel].USDFeeds)
-	server := mockAttestationResponse()
-	defer server.Close()
-	endpoint := server.URL
-	err = DeployCCIPContracts(e.Env, newAddresses, DeployCCIPContractConfig{
-		HomeChainSel:   e.HomeChainSel,
-		FeedChainSel:   e.FeedChainSel,
-		ChainsToDeploy: e.Env.AllChainSelectors(),
-		TokenConfig:    tokenConfig,
-		OCRSecrets:     deployment.XXXGenerateTestOCRSecrets(),
-		USDCConfig: USDCConfig{
-			Enabled: true,
-			USDCAttestationConfig: USDCAttestationConfig{
-				API:         endpoint,
-				APITimeout:  commonconfig.MustNewDuration(time.Second),
-				APIInterval: commonconfig.MustNewDuration(500 * time.Millisecond),
+	USDCCCTPConfig := make(map[cciptypes.ChainSelector]pluginconfig.USDCCCTPTokenConfig)
+	timelocksPerChain := make(map[uint64]*gethwrappers.RBACTimelock)
+	for _, chain := range usdcChains {
+		require.NotNil(t, state.Chains[chain].MockUSDCTokenMessenger)
+		require.NotNil(t, state.Chains[chain].MockUSDCTransmitter)
+		require.NotNil(t, state.Chains[chain].USDCTokenPool)
+		USDCCCTPConfig[cciptypes.ChainSelector(chain)] = pluginconfig.USDCCCTPTokenConfig{
+			SourcePoolAddress:            state.Chains[chain].USDCTokenPool.Address().String(),
+			SourceMessageTransmitterAddr: state.Chains[chain].MockUSDCTransmitter.Address().String(),
+		}
+		timelocksPerChain[chain] = state.Chains[chain].Timelock
+	}
+	var usdcCfg USDCAttestationConfig
+	if isUSDC {
+		server := mockAttestationResponse()
+		defer server.Close()
+		endpoint := server.URL
+		usdcCfg = USDCAttestationConfig{
+			API:         endpoint,
+			APITimeout:  commonconfig.MustNewDuration(time.Second),
+			APIInterval: commonconfig.MustNewDuration(500 * time.Millisecond),
+		}
+	}
+
+	// Deploy second set of changesets to deploy and configure the CCIP contracts.
+	cs02 := []commonchangeset.ChangesetApplication{
+		{
+			Changeset: commonchangeset.WrapChangeSet(DeployChainContracts),
+			Config: DeployChainContractsConfig{
+				ChainSelectors:    allChains,
+				HomeChainSelector: e.HomeChainSel,
 			},
 		},
-	})
+		{
+			Changeset: commonchangeset.WrapChangeSet(InitialAddChain),
+			Config: DeployCCIPContractConfig{
+				HomeChainSel:   e.HomeChainSel,
+				FeedChainSel:   e.FeedChainSel,
+				ChainsToDeploy: allChains,
+				TokenConfig:    tokenConfig,
+				OCRSecrets:     deployment.XXXGenerateTestOCRSecrets(),
+				USDCConfig: USDCConfig{
+					EnabledChains:         usdcChains,
+					USDCAttestationConfig: usdcCfg,
+					CCTPTokenConfig:       USDCCCTPConfig,
+				},
+			},
+		},
+		{
+			Changeset: commonchangeset.WrapChangeSet(CCIPCapabilityJobspec),
+		},
+	}
+
+	e.Env, err = commonchangeset.ApplyChangesets(context.Background(), e.Env, timelocksPerChain, cs02)
```

Comment:
could we inline the list of changeset here? I think it'd be more readable if it was like ApplyChangesets(ctx, e.Env, []common.Changesets{.....})